packages/andromeda-data-storage/src/counter.rs

```diff
@@ -0,0 +1,75 @@
+use andromeda_std::{andr_exec, andr_instantiate, andr_query};
+use cosmwasm_schema::{cw_serde, QueryResponses};
+
+#[andr_instantiate]
+#[cw_serde]
+pub struct InstantiateMsg {
+    pub restriction: CounterRestriction,
```

Comment:
Where is this used in the contract other than saving it during instantiation and `UpdateRestriction`?

Reply:
Only owner/operator can update restriction.
Restriction contains Public and Private.
Only users who have permission can call functions like increment, decrement and so on.
```
pub fn has_permission(
    storage: &dyn Storage,
    addr: &Addr,
) -> Result<bool, ContractError> {
    let is_operator = ADOContract::default().is_owner_or_operator(storage, addr.as_str())?;
    let allowed = match RESTRICTION.load(storage)? {
        CounterRestriction::Private => is_operator,
        CounterRestriction::Public => true,
    };
    Ok(is_operator || allowed)
}
```